### PR TITLE
Enable column edition, and fix ordering

### PR DESCRIPTION
**Pull request by emilienchvt**
This PR enables edition of column names
It also fixes the ordering of the columns in the pipeline view.

Screen recording:
Screen-Recording-2023-07-15-at-1

### CODE REVIEW

File: front/src/modules/ui/board/components/BoardColumn.tsx [hunk 1]
```diff
@@ -26,21 +30,54 @@ export const StyledColumnTitle = styled.h3`
...
+  colorCode,
+  title,
+  onEditTitle,
+  totalAmount,
+  children,
+}: OwnProps) {
+  const [isEditing, setIsEditing] = React.useState(false);
+  const [internalValue, setInternalValue] = React.useState(title);
+
+  function switchEditMode() {
```
Comment (charlesBochet) [inline front/src/modules/ui/board/components/BoardColumn.tsx:56]: toggleEditMode

File: front/src/modules/ui/board/components/BoardColumn.tsx [hunk 2]
```diff
@@ -26,21 +30,54 @@ export const StyledColumnTitle = styled.h3`
...
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setInternalValue(event.target.value);
+    debouncedOnUpdate(event.target.value);
+  };
+
   return (
     <StyledColumn>
-      <StyledHeader>
-        <StyledColumnTitle color={colorCode}>• {title}</StyledColumnTitle>
+      <StyledHeader></StyledHeader>
```
Comment (charlesBochet) [inline front/src/modules/ui/board/components/BoardColumn.tsx]: component is called twice?

File: front/src/modules/ui/board/components/EditColumnTitleInput.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import { useScopedHotkeys } from '@/lib/hotkeys/hooks/useScopedHotkeys';
+import { useSetHotkeyScope } from '@/lib/hotkeys/hooks/useSetHotkeyScope';
+
+import { ColumnHotkeyScope } from './ColumnHotkeyScope';
+
+const StyledEditTitleInput = styled.input`
+  background-color: transparent;
+  border: none;
+  &::placeholder,
+  &::-webkit-input-placeholder {
```
Comment (charlesBochet) [inline front/src/modules/ui/board/components/EditColumnTitleInput.tsx:12]: no need for webkit-input-placeholder?

**Review by charlesBochet — APPROVED** [review]

File: front/src/modules/pipeline-progress/states/boardColumnTotalsFamilySelector.ts
Comment (lucasbordeau) [inline front/src/modules/pipeline-progress/states/boardColumnTotalsFamilySelector.ts:1]: Beautiful

File: front/src/modules/pipeline-progress/states/boardColumnTotalsFamilySelector.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import { selectorFamily } from 'recoil';
+
+import { companyProgressesFamilyState } from '@/companies/states/companyProgressesFamilyState';
+import { BoardPipelineStageColumn } from '@/ui/board/components/Board';
+
+import { boardState } from './boardState';
+
+export const boardColumnTotalsFamilySelector = selectorFamily({
+  key: 'BoardColumnTotalsFamily',
```
Comment (lucasbordeau) [inline front/src/modules/pipeline-progress/states/boardColumnTotalsFamilySelector.ts:9]:
```suggestion
  key: 'boardColumnTotalsFamilySelector',
```

File: front/src/modules/ui/board/components/BoardColumn.tsx [hunk 5]
```diff
@@ -26,21 +30,53 @@ export const StyledColumnTitle = styled.h3`
...
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setInternalValue(event.target.value);
+    debouncedOnUpdate(event.target.value);
+  };
+
   return (
     <StyledColumn>
-      <StyledHeader>
-        <StyledColumnTitle color={colorCode}>• {title}</StyledColumnTitle>
+      <StyledHeader onClick={toggleEditMode}>
```
Comment (lucasbordeau) [inline front/src/modules/ui/board/components/BoardColumn.tsx:68]: Would be nice to have a cursor: pointer

**Review by charlesBochet — APPROVED** [review]

**Pull request merged**